isminet/models/network.py

```
@@ -0,0 +1,177 @@
+        2. Checks that no duplicate VLAN IDs exist in the configuration
+        
+        Raises:
+            ValueError: If VLANs are enabled without configurations or if duplicate VLAN IDs are detected
+        
+        Returns:
+            NetworkConfiguration: The validated network configuration instance
+        """
+        if self.vlan_enabled and not self.vlans:
+            raise ValueError(
+                "VLAN configurations must be provided when VLAN is enabled"
+            )
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)